### PR TITLE
[T5A2][T09-A3]Francis Cheng

### PR DESCRIPTION
**Pull request by francischeng070**
Ready for review

### CODE REVIEW

File: src/seedu/addressbook/commands/Command.java [hunk 1]
```diff
@@ -39,8 +39,13 @@ public static String getMessageForPersonListShownSummary(List<? extends ReadOnly
     /**
      * Executes the command and returns the result.
      */
-    public abstract CommandResult execute();
-
+    public abstract CommandResult execute() throws Exception;
+    
+    /**
+     * To check if the data is modified or not when a command is executed.
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/Command.java:45]: the comment string should use active verb forms. a better comment would be:
`Returns true if the command potentially mutates data`

File: src/seedu/addressbook/logic/Logic.java [hunk 2]
```diff
@@ -85,7 +85,9 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if(command.isMutating()) {
```
Comment (okkhoy) [inline src/seedu/addressbook/logic/Logic.java:88]: you need to update the documentation and the tests to reflect the changed behavior of this method

**Review by okkhoy — COMMENTED** [review]
Notice that your tests fail! this is because, you haven't updated the tests to check for the changed behavior of execute command.

**Comment by okkhoy** [comment]
@francischeng070 
Some comments added. Please ack & close the PR after reading comments.

**Comment by francischeng070** [comment]
ack

**Pull request closed without merging**